frescobaldi_app/vcs/gitrepo.py

```diff
@@ -238,6 +241,8 @@ def branches(self, local=True):
         If local is False also return 'remote' branches.
         """
         res = []
+        #TODO: Are you sure this works asynchronously?
+        # What happpens if this is called while the job is working?
```

Comment:
Right now, `self._update_all_attributes(blocking=True)` is called  synchronously  in `__init__` function.  So `branches()` should be available once the object is constructed. 

`self._local_branches` and `self._remote_branches` should be updated asynchronously when the `repoChanged` emits (Have not implemented).